nixos/modules/services/networking/wireguard.nix

```diff
@@ -153,35 +182,43 @@ let
   wgCommand = "${pkgs.wireguard}/bin/wg";
 
   generateUnit = name: values:
+    assert (values.privateKey != null || values.privateKeyFile != null);
     nameValuePair "wireguard-${name}"
       {
         description = "WireGuard Tunnel - ${name}";
         after = [ "network.target" ];
         wantedBy = [ "multi-user.target" ];
+        script = builtins.concatStringsSep "\n" (lib.flatten([
```

Comment:
Where is the script created and what is its file mode? If it's world readable, then this PR doesn't actually fix anything.